servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -15,351 +15,56 @@
-      } catch (IOException e) {
-        LOG.error(e.getMessage(), e);
-        mWorkerStorage.resetMasterClient();
-        CommonUtils.sleepMs(LOG, Constants.SECOND_MS);
-        cmd = null;
-        int heartbeatTimeout =
-            mTachyonConf.getInt(Constants.WORKER_HEARTBEAT_TIMEOUT_MS, 10 * Constants.SECOND_MS);
-        if (System.currentTimeMillis() - lastHeartbeatMs >= heartbeatTimeout) {
-          throw new RuntimeException("Heartbeat timeout "
-              + (System.currentTimeMillis() - lastHeartbeatMs) + "ms");
-        }
-      }
-
-      if (cmd != null) {
-        switch (cmd.mCommandType) {
-          case Unknown:
-            LOG.error("Unknown command: " + cmd);
-            break;
-          case Nothing:
-            LOG.debug("Nothing command: {}", cmd);
-            break;
-          case Register:
-            LOG.info("Register command: " + cmd);
-            mWorkerStorage.register();
-            break;
-          case Free:
-            mWorkerStorage.freeBlocks(cmd.mData);
-            LOG.info("Free command: " + cmd);
-            break;
-          case Delete:
-            LOG.info("Delete command: " + cmd);
-            break;
-          default:
-            throw new RuntimeException("Un-recognized command from master " + cmd.toString());
+        if (worker != null) {
+          worker.stop();
         }
+      } catch (Exception ex) {
+        LOG.error("Failed to stop block worker. Exiting.", ex);
       }
-
-      mWorkerStorage.checkStatus();
+      System.exit(-1);
     }
+    System.exit(0);
   }
 
   /**
-   * Start the data server thread and heartbeat thread of this TachyonWorker.
-   */
-  public void start() throws IOException {
-    mHeartbeatThread.start();
-    mWorkerMetricsSystem.start();
-    mWebServer.addHandler(mWorkerMetricsSystem.getServletHandler());
-    mWebServer.startWebServer();
-
-    LOG.info("The worker server started @ " + mWorkerAddress);
-    mServer.serve();
-    LOG.info("The worker server ends @ " + mWorkerAddress);
-  }
-
-  /**
-   * Stop this TachyonWorker. Stop all the threads belong to this TachyonWorker.
+   * Verifies that no parameters are passed in
    *
-   * @throws IOException
-   * @throws InterruptedException
+   * @param args command line arguments
    */
-  public void stop() throws IOException, InterruptedException {
-    mStop = true;
-    mWorkerStorage.stop();
-    mWorkerMetricsSystem.stop();
-    mDataServer.close();
-    mServer.stop();
-    mServerTServerSocket.close();
-    mExecutorService.shutdown();
-    while (!mDataServer.isClosed() || mServer.isServing() || mHeartbeatThread.isAlive()) {
-      // TODO The reason to stop and close again is due to some issues in Thrift.
-      mServer.stop();
-      mServerTServerSocket.close();
-      CommonUtils.sleepMs(null, 100);
+  private static void checkArgs(String[] args) {
+    if (args.length != 0) {
+      LOG.info("Usage: java TachyonWorker");
```

Comment:
Isn't it 'LOG.info("Usage: java -cp target/tachyon-" + Version.VERSION + "-jar-with-dependencies.jar tachyon.Worker");` ?